server/fleet/hosts.go

```diff
@@ -321,6 +321,20 @@ type Host struct {
 	LastRestartedAt time.Time `json:"last_restarted_at" db:"last_restarted_at" csv:"last_restarted_at"`
 }
 
+type HostHealth struct {
+	UpdatedAt             time.Time           `json:"updated_at,omitempty" db:"updated_at"`
+	OsVersion             string              `json:"os_version,omitempty" db:"os_version"`
+	DiskEncryptionEnabled *bool               `json:"disk_encryption_enabled,omitempty" db:"disk_encryption_enabled"`
+	VulnerableSoftware    []HostSoftwareEntry `json:"vulnerable_software,omitempty"`
+	FailingPolicies       []*HostPolicy       `json:"failing_policies,omitempty"`
+	Platform              string              `json:"-" db:"platform"`      // Needed to fetch failing policies. Not returned in responses.
+	TeamID                *uint               `json:"team_id" db:"team_id"` // Needed to verify that user can access this host's health data.
```

Comment:
Based on [this comment](https://github.com/fleetdm/fleet/pull/15432#issuecomment-1839522314), it seems like you added this just for authorization and it's not required as part of the API? If so, you should remove it from JSON rendering so it stays a backend-only implementation detail (`json:"-"`).

Reply:
Oh I get it, you need it to be rendered as JSON because the rego authorization uses JSON marshaling for the object. In that case, what you could do is clear it from the API response (after authorization) and add `json":team_id,omitempty"` so that it doesn't show up. The other option I can think of is to use an altogether different struct for authorization, but that seems more trouble than necessary to me.